src/c_api/c_api.cc

```diff
@@ -537,20 +557,18 @@ int MXNDArrayGetShape(NDArrayHandle handle,
   API_END();
 }
 
-int MXNDArrayGetShapeEx(NDArrayHandle handle,
-                        int *out_dim,
-                        const int **out_pdata) {
-  MXAPIThreadLocalEntry *ret = MXAPIThreadLocalStore::Get();
-  API_BEGIN();
-  NDArray *arr = static_cast<NDArray*>(handle);
+template<typename dtype>
+inline void GetShape(NDArrayHandle handle, const dtype** out_pdata, int* out_dim,
```

Comment:
Naming should be consistent with `CreateNDArrayImpl`. If @larroy feels `Impl` is not the best suffix, we can consider `CreateNDArrayHelper` and `GetShapeHelper`. But all these internal template names sholud have consistent styles since we may have many of them.

Reply:
For now GetShape makes sense since this function contains the logic to compute and return shape. So in my opinion keeping it GetShape would be better

Reply:
This is not consistent with CreateNDArrayImpl. Please use consistent naming.

Reply:
CreateNDArrayImpl is not correct naming altogether. There is no point of calling it CreateNDArrayImpl naming should make sense with what is written inside of the function. Impl implies implementation. We are not implementing anything in that function, we are simply calling constructor to create NDArray. I am ok to revert CreateNDArrayImpl -> CreateNDArray. That makes more sense to me. @apeforest do you think this makes sense ?

Reply:
Okay, please revert CreateNDArrayImpl -> CreateNDArray